pkg/gatewayserver/io/basicstationlns/basicstationlns.go

```diff
@@ -185,8 +205,8 @@ func (s *srv) handleTraffic(c echo.Context) error {
 				return
 			case down := <-conn.Down():
 				dnmsg := messages.DownlinkMessage{}
-				//TODO: Add Token check after rebasing https://github.com/TheThingsNetwork/lorawan-stack/pull/589
-				dnmsg.FromDownlinkMessage(ids, *down, 0x00)
+				dlTime := time.Now()
+				dnmsg.FromDownlinkMessage(ids, *down, int64(s.tokens.Next(down.CorrelationIDs, dlTime)), dlTime)
```

Comment:
I would expect a return value and error here.

Is `FromDownlinkMessage()` filling in the pointer receiver? Can we have a conversion function instead? This is essentially similar to UDP translation.